enhancements/network/ovn-multi-external-gateway.md

```diff
@@ -0,0 +1,182 @@
+- Extend test coverage to include the new CRD use cases, matching the same use cases covered in the annotation tests.
+- Support for CRD configuration will be the only mechanism supported in OpenShift 4.14 and onwards. Annotations will still be supported in 4.13, however the CRD will be the only officially supported and documented metohd of configuration.
+- Extend the existing unit tests for the annotation based logic to accommodate for the scenario where CR and annotations exist in 4.13 and onwards. 
+
+### Non-Goals
+
+- Functional improvements on the logic to handle multiple external gateways that are not supported in the existing logic.
+- Removal of the annotation based logic to configure multiple external gateways. This should be carrier out at least no sooner than OpenShift 4.15 to guarantee a stable deprecation path to existing deployments that leverage on the annotations to configure the external gateways.
+
+## Proposal
+
+Create a new CRD that contains a direct relationship to the annotations used in the pod/namespace to manage the external gateways configuration:
+
+Namespace annotations:
+* `k8s.ovn.org/routing-external-gws`
+* `k8s.ovn.org/external-gw-pod-ips`
+
+Pod annotations:
+* `k8s.ovn.org/routing-namespaces`
+* `k8s.ovn.org/routing-network`
+* `k8s.ovn.org/bfd-enabled`
+
+The implementation will also include a new controller that will watch cluster wide for instances of this CRD and apply the changes to the OVN network configuration as it is currently done by the annotation based logic.
+
+The controller will use the OpenShift release version as well as the existence of the annotations in the namespace and pods to apply the changes to the network configuration:
+
+| **OpenShift version** | **Annotations exist** | **CR exists** | **Action**                                        |
+|-----------------------|-----------------------|---------------|---------------------------------------------------|
+| 4.13                  | No                    | Yes           | Use CR information to configure external gateways |
+| 4.13                  | Yes                   | Yes           | Use annotations to configure external gateways    |
+| 4.13                  | Yes                   | No            | Use annotations to configure external gateway     |
+| 4.13                  | No                    | No            | Nothing to do                                     |
+| 4.14                  | No                    | Yes           | Use CR information to configure external gateways |
+| 4.14                  | Yes                   | Yes           | Use CR information to configure external gateways |
+| 4.14                  | Yes                   | No            | Use annotations to configure external gateways    |
+| 4.14                  | No                    | No            | Nothing to do                                     |
```

Comment:
Maybe this is an implementation detail, but how do we plan to do this upstream? We don't have a versioning system there. Assuming this feature is going to go upstream first and then downstream.